docs/public/operators-on-minikube.adoc

```diff
@@ -0,0 +1,204 @@
+
+Discussing this topic is out of scope of this guide so we have stated it as a note.
+====
+
+Enabling the `olm` addon will, among other things, create an `OperatorGroup` in the `operators` namepsace. Make sure that it's there:
+[source,sh]
+----
+$ kubectl get og -n operators
+NAME               AGE
+global-operators   3m37s
+----
+
+If you don't see one, create it using below command:
+[source,sh]
+----
+$ kubectl create -f - << EOF
+apiVersion: operators.coreos.com/v1
+kind: OperatorGroup
+metadata:
+  name: global-operators 
+  namespace: operators 
+spec:
+  targetNamespaces:
+  - operators
+EOF
+----
+
+Now, install the etcd Operator using below command:
+[source,sh]
+----
+$ kubectl create -f - << EOF
+apiVersion: operators.coreos.com/v1alpha1
+kind: Subscription
+metadata:
+  name: etcd
+  namespace: operators
+spec:
+  channel: clusterwide-alpha
+  name: etcd
+  source: operatorhubio-catalog
+  sourceNamespace: olm
+  startingCSV: etcdoperator.v0.9.4-clusterwide
+  installPlanApproval: Automatic
+EOF
+----
+
+Give it a few seconds before checking the availability of the etcd Operator:
+[source,sh]
+----
+$ odo catalog list services
+Operators available in the cluster
+NAME                                CRDs
+etcdoperator.v0.9.4-clusterwide     EtcdCluster, EtcdBackup, EtcdRestore
+----
+
+==== Troubleshooting
+
+If you don't see etcd Operator using above command or by doing `kubectl get csv -n operators`, make sure that pod belonging to the `CatalogSource` named `operatorhubio-catalog` is running:
+[source,sh]
+----
+$ kubectl get po -n olm | grep operatorhubio-catalog
+----
+
+If the state of this pod is `CrashLoopBackOff`, delete it so that Kubernetes will automatically spin up a new pod for the `CatalogSource`:
+
+[source,sh]
+----
+$ kubectl delete po -n olm <name-of-operatorhubio-catalog-pod>
+----
+
+Once the pod for this `CatalogSource` is up, wait a few seconds before trying to find the etcd Operator when you do `odo catalog list services`.
+
+=== Installing the Service Binding Operator
+
+We use the link:https://github.com/redhat-developer/service-binding-operator/[Service Binding Operator] to provide `odo link` feature which links an odo component with an Operator backed service. Thus, to be able to use this feature, it's essential that we install the Operator first.
+
+Service Binding Operator is not yet available via the OLM. The team is link:https://github.com/redhat-developer/service-binding-operator/issues/727[working on making it available] through OLM.
+
+At the moment, to make Service Binding Operator work on Kubernetes, we need to keep a couple of things in mind:
+
+. It needs to be installed by cloning its GitHub repository
+. It must be installed in an individual namespace unlike the cluster wide installation we saw for etcd Operator in previous section
```

Comment:
the operator can be installed in all namespaces.

Reply:
How? Is there a doc for this on SBO's repo or website that a user can refer to?

Reply:
That is the OLM default behaviour, i.e. this is what you get if installed using Subscription (see above). If you want to tell operator to watch only some namespaces, then we need to configure via [`OperatorGroup`](https://docs.openshift.com/container-platform/4.6/operators/understanding/olm/olm-understanding-operatorgroups.html)

Reply:
OK. Maybe a silly question but, for below Operator Group, if I install SBO using the CSV [available from here](https://github.com/operator-framework/community-operators/blob/master/community-operators/service-binding-operator/0.1.1-364/service-binding-operator.v0.1.1-364.clusterserviceversion.yaml), I should be able to see SBO getting installed across all the namespaces, right?
```yaml
apiVersion: operators.coreos.com/v1
kind: OperatorGroup
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"operators.coreos.com/v1","kind":"OperatorGroup","metadata":{"annotations":{},"name":"global-operators","namespace":"operators"}}
    olm.providedAPIs: EtcdBackup.v1beta2.etcd.database.coreos.com,EtcdCluster.v1beta2.etcd.database.coreos.com,EtcdRestore.v1beta2.etcd.database.coreos.com
  name: global-operators
  namespace: operators
spec: {}
status:
  lastUpdated: "2020-10-28T05:58:42Z"
  namespaces:
  - ""
```
The reason I'm expecting it to be available across all namespaces is the way [`installModes` are configured in the CSV](https://github.com/operator-framework/community-operators/blob/master/community-operators/service-binding-operator/0.1.1-364/service-binding-operator.v0.1.1-364.clusterserviceversion.yaml#L181-L189).

I could be wrong in my understanding.

Reply:
Not sure what do you mean installed across all namespaces? Technically, operator gets installed in a given namespace (e..g `operatrors`) but it watches for resources in all namespaces.

Reply:
> Technically, operator gets installed in a given namespace (e..g operatrors) but it watches for resources in all namespaces.

Thanks for this info. I was, tbh, not aware of this. I've opened https://github.com/openshift/odo/issues/4191 to ensure we don't miss this from odo's side.

At the moment, odo requires an Operator to be installed in current working ns to be able to spin up a service from its CR.